app/forms/proof_form.rb

```diff
@@ -1,11 +1,13 @@
 # frozen_string_literal: true
 class ProofForm < ApplicationForm
-  attribute :identity_id
+  attribute :entity_id
   attribute :proof_type_id
   attribute :encryption_fingerprint
   attribute :files
+  attribute :entity_type
 
-  validates :files, :identity_id, :proof_type_id, presence: true
+  validates :files, :entity_id, :proof_type_id, presence: true
+  validates_with EntityTypeValidator
```

Comment:
i don't think we need separate `EntityTypeValidator` class

it only needed in this particular form, so you can inline code here

```ruby
ALLOWED_ENTITY_TYPES = %w[Identity Address].freeze

...

validate :validate_entity_type

...

private

def validate_entity_type
  ...
end
```